Tests/BasicPerformanceTests/SortedArrayPerfTests.swift

```diff
@@ -11,14 +11,44 @@
 import TestSupport
 import Basic
 
+private let magnitude = 5000
+
+private func prng(_ seed: Int) -> Int {
+    return (seed &* 1299827) % magnitude
+}
+
 class SortedArrayPerfTests: XCTestCasePerf {
+
+    private let startSequence = (0..<(magnitude*2)).map(prng)
+    private let overlappingSequence = ((magnitude*1)..<(magnitude*3)).map(prng)
+
+    private var arr: SortedArray<Int> = .init(areInIncreasingOrder: <)
+
+    override func setUp() {
+        arr = .init(areInIncreasingOrder: <)
+        arr.insert(contentsOf: startSequence)
+    }
+
     func testPerformanceOfSortedArrayInAscendingOrder() {
         
         measure() {
-            var arr = SortedArray<Int>(areInIncreasingOrder: <)
             for i in 1...1000 {
-                arr.insert(i)
+                self.arr.insert(i)
             }
         }
     }
+
+    func testPerformanceOfSortedArrayInsertWithDuplicates() {
+        measure() {
+            for element in self.overlappingSequence {
+                self.arr.insert(element)
```

Comment:
The measure block is called 10 times so this will accumulate the result, which may affect the perf numbers. I suggest create an array and then copy it inside the measure block. Something like this:

```swift
// Setup the array.
var initial = SortedArray<Int>(areInIncreasingOrder: <)
initial.append(contentsOf: Array(0..<1000)

let sequenceToInsert = Array(0...<10000)

measure {
    var array = initial
    array.insert(contentsOf: sequenceToInsert)
}
```

We will end up paying the cost of copy-on-write, but thats fine because it should be same for each iteration.